### DISCUSSION TITLE
I3S BSL filter/query/search

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/loaders.gl#2449] [Q&A]
Hi,

Is it possible to filter/query/search in I3S (BSL) data?

Use case:

- input: find all the windows (BSL sublayer) where `attribute (e.g. width)` > 10 => show result in some table => filter sublayer visually (only filtered windows are visible)

I'm not sure if it requires some API endpoint for queries or how it should work. I'm just asking if anybody had similar idea.

Thanks

**Comment by belom88** [comment]
Hi! 
It is possible! We have UI Design for this in [I3S Explorer](https://i3s.loaders.gl/viewer?tileset=turanga-library) (available in [loaders.gl showcase](https://loaders.gl/showcase))
See "Cog" button -> Turanga Library 3 dots button -> Layer Settings
![Screenshot 2023-05-03 at 16 18 18](https://user-images.githubusercontent.com/18549629/235943612-8603025d-8a03-4b7d-9569-6af523aa2c41.png)
We were going to implement filtering by Floor and construction stage but now this feature is on hold.
To filter by attribute it is necessary to load feature attributes that are stored by [I3S Spec](https://github.com/Esri/i3s-spec) and post-process mesh data (some new logic in @loaders.gl / @deck.gl is required)
